pkgs/development/python-modules/posting/default.nix

```
@@ -0,0 +1,130 @@
+{
+  buildPythonPackage,
+  hatchling,
+  lib,
+  fetchFromGitHub,
+  fetchPypi,
+  pydantic,
+  poetry-core,
+  rich,
+  platformdirs,
+  typing-extensions,
+  linkify-it-py,
+  httpx,
+  click,
+  xdg-base-dirs,
+  click-default-group,
+  pyperclip,
+  pyyaml,
+  pydantic-settings,
+  python-dotenv,
+  watchfiles,
+  toml,
+  typer,
+}:
+let
+  openapi-pydantic = buildPythonPackage rec {
+    pname = "openapi_pydantic";
+    version = "0.5.0";
+
+    src = fetchPypi {
+      inherit pname version;
+      sha256 = "sha256-pI+I4pBKBW4e9tRyjPsvNqoyE84ZT7CfwEJZuQBxZfA=";
+    };
+
+    propagatedBuildInputs = [
+      pydantic
+    ];
+
+    pyproject = true;
+    nativeBuildInputs = [
+      poetry-core
+    ];
+  };
+
+  textual-2_1_2 = buildPythonPackage rec {
+    pname = "textual";
+    version = "2.1.2";
+
+    src = fetchPypi {
+      inherit pname version;
+      sha256 = "sha256-quP5/eAMdEC+AOPDrBieAtAU9SmK/cMhMvk0gPngkUY=";
+    };
+
+    propagatedBuildInputs = [
+      rich
+      platformdirs
+      typing-extensions
+      linkify-it-py
+    ];
+
+    pyproject = true;
+    nativeBuildInputs = [
+      poetry-core
+    ];
+  };
+
+  textual-autocomplete = buildPythonPackage rec {
+    pname = "textual_autocomplete";
+    version = "4.0.0a0";
+
+    src = fetchPypi {
+      inherit pname version;
+      sha256 = "sha256-wsjmgODvFgfbyqxW3jsH88JC8z0TZQOChLgics7wAHY=";
+    };
+
+    propagatedBuildInputs = [
+      textual-2_1_2
+    ];
+
+    pyproject = true;
+    nativeBuildInputs = [
+      hatchling
+    ];
+  };
+in
+buildPythonPackage rec {
+  pname = "posting";
+  version = "2.5.4";
+
+  src = fetchFromGitHub {
+    owner = "darrenburns";
+    repo = "posting";
+    rev = version;
```

```suggestion
    tag = version;
```
Is the new preferred method to fetch a tag